modin/numpy/arr.py

```diff
@@ -1469,6 +1559,283 @@ def sum(
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        result = self._query_compiler._logical_not()
+        return fix_dtypes_and_determine_return(result, self._ndim, dtype, out, where)
+
+    def _logical_binop(
+        self, qc_method_name, x2, out, where, casting, order, dtype, subok
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        if is_scalar(x2):
+            return fix_dtypes_and_determine_return(
+                getattr(self, qc_method_name)(x2),
+                self._ndim,
+                dtype,
+                out,
+                where,
+            )
+        caller, callee, new_ndim, kwargs = self._binary_op(x2)
+        # breakpoint()
+        if caller._query_compiler != self._query_compiler:
+            # In this case, we are doing an operation that looks like this 1D_object,op(2D_object).
+            # For Modin to broadcast directly, we have to swap it so that the operation is actually
+            # 2D_object.op(1D_object).
+            caller, callee = callee, caller
+        # If the LHS is 2D and the RHS is 1D, we need to convert the RHS into a dataframe instead of a series
+        # because numpy would otherwise think one of the objects is a series
+        # if caller._ndim == 2 and callee._ndim == 1:
+        #     callee = array(pd.DataFrame(query_compiler=callee._query_compiler).transpose())
+        result = getattr(caller._query_compiler, qc_method_name)(
+            callee._query_compiler, **kwargs
+        )
+        return fix_dtypes_and_determine_return(result, new_ndim, dtype, out, where)
+
+    def _logical_and(
+        self,
+        x2,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        return self._logical_binop(
+            "_logical_and", x2, out, where, casting, order, dtype, subok
+        )
+
+    def _logical_or(
+        self,
+        x2,
+        /,
+        out=None,
+        *,
+        where=True,
+        casting="same_kind",
+        order="K",
+        dtype=None,
+        subok=True,
+    ):
+        check_kwargs(where=where, casting=casting, order=order, subok=subok)
+        if is_scalar(x2):
+            return fix_dtypes_and_determine_return(
+                self._query_compiler._logical_or(x2),
+                self._ndim,
+                dtype,
+                out,
+                where,
+            )
+        caller, callee, new_ndim, _ = self._binary_op(x2)
+        if caller._query_compiler != self._query_compiler:
+            # In this case, we are doing an operation that looks like this 1D_object | 2D_object.
+            # For Modin to broadcast directly, we have to swap it so that the operation is actually
+            # 2D_object.or(1D_object).
+            caller, callee = callee, caller
```

Comment:
Why don't we use logical bin op here?

Reply:
Seems like I forgot to change it, will fix.